File: java-example/src/test/java/org/onflow/examples/java/getAccount/GetAccountAccessAPIConnectorTest.java
package org.onflow.examples.java.getAccount;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowServiceAccountCredentials;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.common.test.TestAccount;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowAccount;
import org.onflow.flow.sdk.FlowAddress;

import java.math.BigDecimal;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class GetAccountAccessAPIConnectorTest {
    @FlowServiceAccountCredentials
    private TestAccount testAccount;
    @FlowTestClient
    private FlowAccessApi accessAPI;
    private GetAccountAccessAPIConnector accountConnector;

    @BeforeEach
    public void setup() {
        accountConnector = new GetAccountAccessAPIConnector(accessAPI);
    }

    @Test
    public void testCanFetchAccountFromLatestBlock() {
        FlowAddress address = testAccount.getFlowAddress();
        FlowAccount account = accountConnector.getAccountAtLatestBlock(address);

        Assertions.assertNotNull(account, "Account should not be null");
        Assertions.assertEquals(address.getBase16Value(), account.getAddress().getBase16Value(), "Address should match");
        Assertions.assertTrue(account.getBalance().compareTo(BigDecimal.ZERO) >= 0, "Account balance should be non-negative");
    }

    @Test
    public void testCanFetchAccountFromBlockByHeight0() {
        FlowAddress address = testAccount.getFlowAddress();
        FlowAccount account = accountConnector.getAccountAtBlockHeight(address, 0);

        Assertions.assertNotNull(account, "Account should not be null");
        Assertions.assertEquals(address.getBase16Value(), account.getAddress().getBase16Value(), "Address should match");
        Assertions.assertTrue(account.getBalance().compareTo(BigDecimal.ZERO) >= 0, "Account balance should be non-negative");
    }

    @Test
    public void testCanFetchAccountBalance() {
        FlowAddress address = testAccount.getFlowAddress();
        BigDecimal balance = accountConnector.getAccountBalance(address);

        Assertions.assertNotNull(balance, "Balance should not be null");
        Assertions.assertTrue(balance.compareTo(BigDecimal.ZERO) >= 0, "Balance should be non-negative");
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/getAccount/GetAccountAccessAPIConnectorTest.kt
package org.onflow.examples.kotlin.getAccount

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.flow.common.test.*
import org.onflow.flow.sdk.FlowAccessApi
import java.math.BigDecimal

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class GetAccountAccessAPIConnectorTest {

    @FlowServiceAccountCredentials
    lateinit var testAccount: TestAccount

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var accountConnector: GetAccountAccessAPIConnector

    @BeforeEach
    fun setup() {
        accountConnector = GetAccountAccessAPIConnector(accessAPI)
    }

    @Test
    fun `Can fetch account from the latest block`() {
        val address = testAccount.flowAddress
        val account = accountConnector.getAccountAtLatestBlock(address)

        assertNotNull(account, "Account should not be null")
        assertEquals(address.base16Value, account.address.base16Value, "Address should match")
        assertTrue(account.balance >= BigDecimal.ZERO, "Account balance should be non-negative")
    }

    @Test
    fun `Can fetch account from block by height 0`() {
        val address = testAccount.flowAddress
        val account = accountConnector.getAccountAtBlockHeight(address, 0)

        assertNotNull(account, "Account should not be null")
        assertEquals(address.base16Value, account.address.base16Value, "Address should match")
        assertTrue(account.balance >= BigDecimal.ZERO, "Account balance should be non-negative")
    }

    @Test
    fun `Can fetch account balance`() {
        val address = testAccount.flowAddress
        val balance = accountConnector.getAccountBalance(address)

        assertNotNull(balance, "Balance should not be null")
        assertTrue(balance >= BigDecimal.ZERO, "Balance should be non-negative")
    }
}
